### PR TITLE
Nisha | Test | Version 3 & Version 4 Group & Transaction wise Export

### PR DESCRIPTION
**Pull request by nisha-walkover**
https://app.clickup.com/t/86cvuhebx
https://app.clickup.com/t/86cvuhedt

### CODE REVIEW

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 1]
```diff
@@ -105,4 +105,82 @@ export class InventoryComponentStore extends ComponentStore<any> {
...
+     * @memberof InventoryComponentStore
+     */
+    readonly exportGroup = this.effect((data$: Observable<{ stockReportRequest: any, queryParams: any }>) => {
+        return data$.pipe(
+            switchMap((req) => {
+                this.patchState({ isLoading: true });
+                return this.inventoryService.getGroupWiseReportExport(req.queryParams, req.stockReportRequest).pipe(
+                    tapResponse(
+                        (res: BaseResponse<any, any>) => {
+                            if (res.status === "success") {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]:
```suggestion
                            if (res.status === "success" && typeof res?.body === "string") {
```

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 2]
```diff
@@ -105,4 +105,82 @@ export class InventoryComponentStore extends ComponentStore<any> {
...
+                                if (typeof res?.body === "string") {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]:
```suggestion
                           
```

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 3]
```diff
@@ -105,4 +105,82 @@ export class InventoryComponentStore extends ComponentStore<any> {
...
+                                    this.toaster.showSnackBar("success", res.body);
+                                    this.router.navigate(["/pages/downloads"]);
+                                    return this.patchState({
+                                        isLoading: false
+                                    });
+                                }
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]:
```suggestion

```

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts [hunk 4]
```diff
@@ -105,4 +105,82 @@ export class InventoryComponentStore extends ComponentStore<any> {
...
+     * @memberof InventoryComponentStore
+     */
+    readonly exportTransaction = this.effect((data$: Observable<{ stockReportRequest: any, queryParams: any }>) => {
+        return data$.pipe(
+            switchMap((req) => {
+                this.patchState({ isLoading: true });
+                return this.inventoryService.getTransactionReportExport(req.queryParams, req.stockReportRequest).pipe(
+                    tapResponse(
+                        (res: BaseResponse<any, any>) => {
+                            if (res.status === "success") {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]:
```suggestion
                            if (res.status === "success" && typeof res?.body === "string") {
```

Comment (dvCodeWorld) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: and remove below if condition

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts [hunk 5]
```diff
@@ -315,63 +315,91 @@ export class ReportFiltersComponent implements OnInit, OnChanges, OnDestroy {
...
-            else if (column === 'inward_amount') {
-                this.stockReportRequestExport.showInwardsValue = true;
-            }
             else if (column === 'outward_quantity') {
                 this.stockReportRequestExport.showOutwardsQty = true;
             }
-            else if (column === 'outward_amount') {
-                this.stockReportRequestExport.showOutwardsValue = true;
+            /* for value filter selected in item, variant & group */
+            if (this.searchPage === InventoryReportType.stock || InventoryReportType.variant || InventoryReportType.group) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: check this once

if (this.searchPage === InventoryReportType.stock || this.searchPage ===  InventoryReportType.variant || this.searchPage ===  InventoryReportType.group) {

aise to nhi likhna chah rhe the ?

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts [hunk 6]
```diff
@@ -315,63 +315,91 @@ export class ReportFiltersComponent implements OnInit, OnChanges, OnDestroy {
...
+                    this.stockReportRequestExport.showClosingStockQty = true;
+                }
+                else if (column === 'closing_amount') {
+                    this.stockReportRequestExport.showClosingStockValue = true;
+                }
             }
-            else if (column === 'closing_quantity') {
-                this.stockReportRequestExport.showClosingStockQty = true;
+            /* for value filter selected in item, variant & transaction */
+            if (this.searchPage === InventoryReportType.stock || InventoryReportType.variant || InventoryReportType.transaction) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]:
```suggestion
            if (this.searchPage === InventoryReportType.stock || InventoryReportType.variant || InventoryReportType.transaction && column === 'stock_name') {
```

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts [hunk 7]
```diff
@@ -315,63 +315,91 @@ export class ReportFiltersComponent implements OnInit, OnChanges, OnDestroy {
...
+                if (column === 'stock_name') {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]:
```suggestion

```

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts [hunk 8]
```diff
@@ -315,63 +315,91 @@ export class ReportFiltersComponent implements OnInit, OnChanges, OnDestroy {
...
+                    this.stockReportRequestExport.showStockName = true;
+                }
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]:
```suggestion

```

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts [hunk 9]
```diff
@@ -315,63 +315,91 @@ export class ReportFiltersComponent implements OnInit, OnChanges, OnDestroy {
...
             }
-            else if (column === 'closing_amount') {
-                this.stockReportRequestExport.showClosingStockValue = true;
+            /* for value filter selected in item & variant both */
+            if (this.searchPage === InventoryReportType.stock || InventoryReportType.variant) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]:
```suggestion
            if (this.searchPage === InventoryReportType.stock || InventoryReportType.variant && column === 'unit_name') {
```

Comment (dvCodeWorld) [reply apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: remove below if block

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts [hunk 10]
```diff
@@ -315,63 +315,91 @@ export class ReportFiltersComponent implements OnInit, OnChanges, OnDestroy {
...
+                if (column === 'unit_name') {
+                    this.stockReportRequestExport.showUnitName = true;
+                }
             }
-            /* for value filter selected in variant */
-            if (InventoryReportType.variant) {
-                if(column === 'variant_name') {
+            /* for value filter selected in variant & transaction both */
+            if (this.searchPage === InventoryReportType.variant || InventoryReportType.transaction) {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]:
```suggestion
            if (this.searchPage === InventoryReportType.variant || InventoryReportType.transaction && column === 'variant_name') {
```

Comment (dvCodeWorld) [reply apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: remove below if block

File: apps/web-giddh/src/app/services/inventory.service.ts [hunk 11]
```diff
@@ -1213,6 +1213,30 @@ export class InventoryService {
...
+     * @param {*} queryParams
+     * @param {InventoryReportRequestExport} stockReportRequest
+     * @return {*}  {Observable<BaseResponse<InventoryReportRequestExport, InventoryReportRequest>>}
+     * @memberof InventoryService
+     */
+    public getTransactionReportExport(queryParams: any, stockReportRequest: InventoryReportRequestExport): Observable<BaseResponse<InventoryReportRequestExport, InventoryReportRequest>> {
+            this.companyUniqueName = this.generalService.companyUniqueName;
+            return this.http.post(this.config.apiUrl + INVENTORY_API.INVENTORY_TRANSACTION_EXPORT
+                ?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))
+                ?.replace(':from', encodeURIComponent(queryParams?.from))
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/inventory.service.ts]:
```suggestion
                ?.replace(':from', encodeURIComponent(queryParams?.from ?? ''))
```

File: apps/web-giddh/src/app/services/inventory.service.ts [hunk 12]
```diff
@@ -1213,6 +1213,30 @@ export class InventoryService {
...
+                ?.replace(':to', encodeURIComponent(queryParams?.to))
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/inventory.service.ts]:
```suggestion
                ?.replace(':to', encodeURIComponent(queryParams?.to ?? ''))
```

File: apps/web-giddh/src/app/services/inventory.service.ts [hunk 13]
```diff
@@ -1253,6 +1277,30 @@ export class InventoryService {
...
+     * @param {*} queryParams
+     * @param {InventoryReportRequestExport} stockReportRequest
+     * @return {*}  {Observable<BaseResponse<InventoryReportRequestExport, InventoryReportRequest>>}
+     * @memberof InventoryService
+     */
+    public getGroupWiseReportExport(queryParams: any, stockReportRequest: InventoryReportRequestExport): Observable<BaseResponse<InventoryReportRequestExport, InventoryReportRequest>> {
+        this.companyUniqueName = this.generalService.companyUniqueName;
+        return this.http.post(this.config.apiUrl + INVENTORY_API.INVENTORY_GROUP_WISE_EXPORT
+            ?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))
+            ?.replace(':from', encodeURIComponent(queryParams?.from))
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/inventory.service.ts]:
```suggestion
            ?.replace(':from', encodeURIComponent(queryParams?.from ?? ''))
```

File: apps/web-giddh/src/app/services/inventory.service.ts [hunk 14]
```diff
@@ -1253,6 +1277,30 @@ export class InventoryService {
...
+            ?.replace(':to', encodeURIComponent(queryParams?.to))
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/inventory.service.ts]:
```suggestion
            ?.replace(':to', encodeURIComponent(queryParams?.to ?? ''))
```

File: apps/web-giddh/src/app/services/inventory.service.ts [hunk 15]
```diff
@@ -1375,6 +1399,30 @@ export class InventoryService {
...
+     * @param {*} queryParams
+     * @param {InventoryReportRequestExport} stockReportRequest
+     * @return {*}  {Observable<BaseResponse<InventoryReportRequestExport, InventoryReportRequest>>}
+     * @memberof InventoryService
+     */
+    public getVariantWiseReportExport(queryParams: any, stockReportRequest: InventoryReportRequestExport): Observable<BaseResponse<InventoryReportRequestExport, InventoryReportRequest>> {
+        this.companyUniqueName = this.generalService.companyUniqueName;
+        return this.http.post(this.config.apiUrl + INVENTORY_API.INVENTORY_VARIANT_WISE_EXPORT
+            ?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))
+            ?.replace(':from', encodeURIComponent(queryParams?.from))
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/inventory.service.ts]:
```suggestion
            ?.replace(':from', encodeURIComponent(queryParams?.from ?? ''))
```

File: apps/web-giddh/src/app/services/inventory.service.ts [hunk 16]
```diff
@@ -1375,6 +1399,30 @@ export class InventoryService {
...
+            ?.replace(':to', encodeURIComponent(queryParams?.to))
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/inventory.service.ts]:
```suggestion
            ?.replace(':to', encodeURIComponent(queryParams?.to ?? ''))
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts (hunk 5 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: no
done

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 4 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 3 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 2 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/inventory.store.ts (hunk 1 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/inventory.store.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts (hunk 6 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts (hunk 7 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts (hunk 8 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts (hunk 9 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: done

File: apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts (hunk 10 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/new-inventory/component/report-filters/report-filters.component.ts]: done

File: apps/web-giddh/src/app/services/inventory.service.ts (hunk 11 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/inventory.service.ts]: done

File: apps/web-giddh/src/app/services/inventory.service.ts (hunk 12 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/inventory.service.ts]: done

File: apps/web-giddh/src/app/services/inventory.service.ts (hunk 13 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/inventory.service.ts]: done

File: apps/web-giddh/src/app/services/inventory.service.ts (hunk 14 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/inventory.service.ts]: done

File: apps/web-giddh/src/app/services/inventory.service.ts (hunk 15 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/inventory.service.ts]: done

File: apps/web-giddh/src/app/services/inventory.service.ts (hunk 16 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/inventory.service.ts]: done

**Review by dvCodeWorld — APPROVED** [review]

**Pull request merged**